### PR TITLE
Prepare xparse for ltcmd

### PR DESCRIPTION
**Pull request by PhelypeOleinik**
This PR is companion to the [`cmd` branch](https://github.com/latex3/latex2e/commits/cmd) in the LaTeX2e repository.

This PR freezes `xparse` in this repository, so that development continues as `ltcmd` in `latex2e`.  This PR can be released to CTAN without the `latex2e` part being released yet, so this has enough time to be on people's distributions before the LaTeX2e release.

All the `xparse` sources were removed from `xparse.dtx`, except for the deprecated argument types, and except for documentation, which is left so that `texdoc xparse` returns the expected until the LaTeX2e release.

### CODE REVIEW

**Comment by josephwright** [comment]
Perhaps I miss something, but the `xparse` documentation is all still there: shouldn't almost all of it be gone?

**Comment by PhelypeOleinik** [comment]
What I understood the other day from our conversation at the tex.sx chatroom, the documentation should so that `texdoc xparse` returns the expected until the LaTeX2e release, when we will have `userguide3`, then the documentation here can be removed, and add a text that points to `userguide3`.  I can remove of course, but I think it would be bad to just kill `texdoc xparse` and have no document to point users to...

**Comment by josephwright** [comment]
OK, so long as it's in hand

File: l3packages/xparse/xparse.ins [hunk 1]
```diff
@@ -42,8 +42,7 @@ and all files in that bundle must be distributed together.
 
 \keepsilent
 
-\generate{\file{xparse-generic.tex}{\from{xparse.dtx}{core}}
-          \file{xparse.sty}{\from{xparse.dtx}{package}}
+\generate{\file{xparse.sty}{\from{xparse.dtx}{package}}
           \file{xparse.ltx}{\from{xparse.dtx}{2ekernel}}}
```
Comment (FrankMittelbach) [inline l3packages/xparse/xparse.ins:46]: what kind of 2ekernel is still produced here? (or is that just temp while the master kernel has different code?)

**Review by FrankMittelbach — APPROVED** [review]
I haven't actually tried to run it, but looking over the files (briefly) I would say give it a try.
But it is of course essential to fully test it together with 2e master before anything gets uploaded to CTAN.

File: l3packages/xparse/xparse.ins (hunk 1 above)
Comment (PhelypeOleinik) [reply l3packages/xparse/xparse.ins:46]: `xparse.ltx` is currently needed to build the 2020-10-01 LaTeX format, so we can't take it out just yet.

**Comment by PhelypeOleinik** [comment]
@FrankMittelbach I tested as extensively as I could with:
 - LaTeX pre-2020-10-01: no `xparse` preloaded
 - LaTeX 2020-10-01: `xparse` preloaded in the kernel (and oh boy, this intermediate step was _lots_ of trouble)
 - LaTeX 2021-05-01: `xparse` built into the kernel as `ltcmd`

In these three cases I also tested `latexrelease` rolling backwards and forwards to each other (some don't work because of other problems (_that_ one) with `latexrelease`), and they worked, and also doing `\usepackage{xparse}` worked in all cases.  You can find the test file I used [here](https://github.com/latex3/latex2e/blob/cmd/base/testfiles-disabled/ltcmd-latexrelease.lvt).

So the tests were pretty extensive, but they could not be fully automated because it required rebuilding the LaTeX format in different points in time, so a lot of manual intervention, which means enough room for errors, so if someone wants to test too, to make sure, that would be good.  To test you just need to `git checkout cmd`, then `l3build install` in `l3packages/xparse` and this version of `xparse` should be available.

File: l3packages/xparse/xparse.ins (hunk 1 above)
Comment (FrankMittelbach) [reply l3packages/xparse/xparse.ins:46]: that's what I thought (hoped :-)) but I couldn't check easily

**Pull request merged**